src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/listingrequest/UpdateListingRequestAction.java

```diff
@@ -136,40 +136,85 @@ public UUID getUserTransactionId() {
 
     @Override
     public UUID[] getAssociatedEntitiesIds() {
-        val ids = hearingParts
+        val ids = hearingPartsToAdd
             .stream()
             .map(HearingPart::getId)
             .collect(Collectors.toList());
 
-        ids.addAll(hearingParts
+        ids.addAll(hearingPartsToRemove
             .stream()
-            .filter(Objects::nonNull)
-            .map(HearingPart::getSessionId)
-            .collect(Collectors.toList()));
+            .map(HearingPart::getId)
+            .collect(Collectors.toList())
+        );
+
+        if (hearing.getStatus().getStatus().equals(Status.Listed)) {
+            ids.addAll(hearingPartsToRemove
+                .stream()
+                .filter(Objects::nonNull)
+                .map(HearingPart::getSessionId)
+                .collect(Collectors.toList()));
+        }
 
         ids.add(updateListingRequest.getId());
 
         return ids.stream().toArray(UUID[]::new);
     }
 
-    private void getHearingPartsWithStatus(Status status) {
-        switch (status) {
-            case Listed:
-                hearingParts = hearing.getHearingParts()
-                    .stream()
-                    .filter(hp -> hp.getStatus().getStatus().equals(status))
-                    .sorted(Comparator.comparing(HearingPart::getStart))
-                    .collect(Collectors.toList());
-                break;
-            case Unlisted:
-                hearingParts = hearing.getHearingParts()
-                    .stream()
-                    .filter(hp -> hp.getStatus().getStatus().equals(status))
-                    .collect(Collectors.toList());
-                break;
-            default:
-                break;
+    private void getUnlistedHearingParts() {
+        hearingParts = hearing.getHearingParts()
+            .stream()
+            .filter(hp -> hp.getStatus().getStatus().equals(Status.Unlisted))
+            .collect(Collectors.toList());
+    }
+
+    private void getListedHearingParts() {
+        hearingParts = hearing.getHearingParts()
+            .stream()
+            .filter(hp -> hp.getStatus().getStatus().equals(Status.Listed))
+            .sorted(Comparator.comparing(HearingPart::getStart))
+            .collect(Collectors.toList());
+    }
+
+    private void getHearingParts() {
+        int diff = updateListingRequest.getNumberOfSessions() - hearing.getNumberOfSessions();
+        if (diff > 0) {
+            addHearingParts(diff);
+        } else if (diff < 0) {
+            removeHearingParts(Math.abs(diff));
+        }
+    }
+
+    private void addHearingParts(int numberOfPartsToAdd) {
```

Comment:
I know it's a copy of what was here already, but in my understanding, this method doesn't add an integer it creates HPs and adds to hearing.... maybe just addNewHearingParts will be better, though it's still potentially confusing (for me)